### ISSUE TITLE
Property doesn't render correctly if property, used as a part of it is mapped to the label

### ISSUE THREAD

**Issue by mishamyte** [serilog-contrib/serilog-sinks-grafana-loki#138]
### Discussed in https://github.com/serilog-contrib/serilog-sinks-grafana-loki/discussions/137

<div type='discussions-op-text'>

<sup>Originally posted by **Depechie** October  7, 2022</sup>
Was wondering if we are doing something wrong, or misjudging the features?
Currently we use Serilog to format our log entries and use the loki sink to push it in json format.

If we look at a result in Loki we get this

<img width="744" alt="Screenshot 2022-10-07 155644" src="https://user-images.githubusercontent.com/351693/194571452-9632c6b4-c10d-4b4b-8862-9b10eb70c0c5.png">

If we now use the propertiesAsLabel feature we get following result

<img width="746" alt="Screenshot 2022-10-07 160254" src="https://user-images.githubusercontent.com/351693/194572236-1a7283f4-ee32-490d-9e2f-62685ad78403.png">

So the actual properties are indeed transformed, but that seems to also impact the payload structure... the Message line now no longer maps the data but only show the placeholders, like {RequestPath}.

Is this expected behaviour?</div>

### Which version of Serilog.Sinks.Grafana.Loki are you using?
v8.0.0

**Comment by mishamyte** [comment]
The problem is `TextFormatter` gets `LogEvent` without properties, that will be mapped to labels. This behavior exists because we don't want to create duplicates (properties, which would be both labels and body rendering).

So the way should be found how pass it to formatter and use for message rendering, but not render as a part of the body

**Comment by mishamyte** [comment]
Seems like it is not possible to do with current abstractions, should be a point of review and breaking changes

**Comment by EraYaN** [comment]
Honestly for our use case leaving the labels as properties (and thus having a "duplicate" would be an acceptable solution. At least one we could opt in too. Basically leave them in the property list and just extract them towards labels, it's how it used to work for v7 and earlier it seems.

**Comment by mishamyte** [comment]
> Honestly for our use case leaving the labels as properties (and thus having a "duplicate") would be an acceptable solution. At least one we could opt in too. Basically leave them in the property list and just extract them towards labels, it's how it used to work for v7 and earlier it seems.
> 
> Say an option that opts out of this partitioning: https://github.com/serilog-contrib/serilog-sinks-grafana-loki/blob/master/src/Serilog.Sinks.Grafana.Loki/LokiBatchFormatter.cs#L185

Yep, this is the problem. To formatter we pass the properties, were not mapped to labels. So while formatter tries to render a `Message` (for example), rendering input doesn't consist of all needed tokens for rendering. 

This is a reason why it is not possible without breaking contracts or scary workarounds fix it in v8. 

But this is the point of fix for v9

**Comment by EraYaN** [comment]
I'll whip up a quick fork for our internal use that could be released in the v8 line of versions if you want. It's a bit of a workaround (the opt-out) but it also doesn't break anything.

**Comment by mishamyte** [comment]
Yep, see your solution. Seems nice as a workaround for now. Let me think

**Comment by EraYaN** [comment]
Cool, if you need me to rename the parameter or anything like that let me know.

**Comment by mishamyte** [comment]
Thanks for your work, @EraYaN!

I think we should add it to the current versions as a workaround for the situation. Will prepare the release

**Comment by mishamyte** [comment]
Thanks once again!

Released as [v8.1.0](https://www.nuget.org/packages/Serilog.Sinks.Grafana.Loki/8.1.0)

**Comment by mishamyte** [comment]
Properly fixed in [v9.0.0](https://www.nuget.org/packages/Serilog.Sinks.Grafana.Loki/9.0.0) (the v8.1.0 change was a workaround) - properties promoted to labels now also stay in the log body, and the message template is never modified.

Note this is a behavior change - see [Upgrading from v8 to v9](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/wiki/Upgrading-from-v8-to-v9). The rewrite also made the sink ~55× lighter on allocations and ~4.5× faster on the common workload ([Benchmarks](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/wiki/Benchmarks)).
